src/core/structures/PDFOutlines.ts

```diff
@@ -0,0 +1,178 @@
+import PDFDict, { DictMap } from 'src/core/objects/PDFDict';
+import PDFName from 'src/core/objects/PDFName';
+import PDFNumber from 'src/core/objects/PDFNumber';
+import PDFString from 'src/core/objects/PDFString';
+import PDFRef from 'src/core/objects/PDFRef';
+import PDFContext from 'src/core/PDFContext';
+import PDFPage from 'src/api/PDFPage';
+import { InvalidTargetIndexError } from 'src/core/errors';
+
+export type OutlineNode = PDFOutlines;
+
+export interface outlineOptions {
+  expanded?: boolean;
+  page?: PDFPage;
+}
+
+class PDFOutlines extends PDFDict {
+  static withContext = (
+    context: PDFContext,
+    options?: outlineOptions,
+    title?: string,
+    parent?: PDFRef,
+  ) => {
+    const dict = new Map();
+    if (!parent) {
+      dict.set(PDFName.Type, PDFName.Outlines);
+    } else {
+      dict.set(PDFName.Title, title ? PDFString.of(title) : title);
```

Comment:
https://github.com/Hopding/pdf-lib/issues/516#issuecomment-656958214

Reply:
Setting the outline title with to a string with an unmatched parentheses causes a corrupt PDF.
Per #516 , using PDFHexString appears to fix this problem.

https://github.com/divinehawk/pdf-lib/commit/57ace824d3492bae1705818713a7a1dbc8a80c06